### PR TITLE
doc(stoneintg-756): write how to manually re-trigger integration test

### PR DESCRIPTION
**Pull request by 14rcole**
Add a document that describes the procedure for manually re-triggering an integration test.

### CODE REVIEW

File: docs/modules/ROOT/pages/how-to-guides/testing_applications/retriggering_tests.adoc [hunk 1]
```diff
@@ -0,0 +1,13 @@
+= Retriggering Tests
```
Comment (MartinBasti) [inline docs/modules/ROOT/pages/how-to-guides/testing_applications/retriggering_tests.adoc:1]:
```suggestion
= Retriggering Integration Tests
```

Comment (14rcole) [reply docs/modules/ROOT/pages/how-to-guides/testing_applications/retriggering_tests.adoc]: Fixed, I also renamed the file to match the new title.

**Review by dirgim — APPROVED** [review]
LGTM

**Review by kasemAlem — APPROVED** [review]

**Comment by ralphbean** [comment]
@14rcole this needs to be added to some index higher in the git repo so that it will appear in the navigation for users.

**Comment by ralphbean** [comment]
Yeah, see `docs/modules/ROOT/nav-how-to-guides.adoc`. I think you need to add a xref link to that list so that this new doc is findable.

File: docs/modules/ROOT/pages/how-to-guides/testing_applications/retriggering_integration_tests.adoc [hunk 2]
```diff
@@ -0,0 +1,13 @@
+= Retriggering Integration Tests
+
+Integration test scenarios for a given snapshot can be re-triggered by adding a label to the snapshot.
+
+== Prerequisites
+The snapshot for which tests will be retriggered has completed its initial tests
+
```
Comment (ralphbean) [inline docs/modules/ROOT/pages/how-to-guides/testing_applications/retriggering_integration_tests.adoc:7]: Add CLI access as a pre-requisite and link to the doc about how to get CLI access with `xref`.

File: docs/modules/ROOT/pages/how-to-guides/testing_applications/retriggering_integration_tests.adoc [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+== Procedure
+. Determine which IntegrationTestScenario should be re-run.  For a given snapshot, only one scenario can
+be re-run at a time.
+. Add the label `test.appstudio.openshift.io/run` to the snapshot. Set the value to the name of the scenario to be re-run
```
Comment (ralphbean) [inline docs/modules/ROOT/pages/how-to-guides/testing_applications/retriggering_integration_tests.adoc:11]: Here, say something about "use kubectl edit" to add a label?

File: docs/modules/ROOT/pages/how-to-guides/testing_applications/retriggering_integration_tests.adoc (hunk 2 above)
Comment (ralphbean) [reply docs/modules/ROOT/pages/how-to-guides/testing_applications/retriggering_integration_tests.adoc:7]: I think the one you want is `getting-started/getting_started_in_cli.adoc`.

File: docs/modules/ROOT/pages/how-to-guides/testing_applications/retriggering_integration_tests.adoc [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+. The tests will be re-triggered automatically.  Once they are re-triggered the label will be removed.  At this point you
```
Comment (ralphbean) [inline docs/modules/ROOT/pages/how-to-guides/testing_applications/retriggering_integration_tests.adoc:12]: Here maybe say "use `kubectl get pipelineruns`" or `tkn pr list` to confirm that your test pipelinerun was successfully re-triggered"?

File: docs/modules/ROOT/nav-how-to-guides.adoc [hunk 5]
```diff
@@ -13,6 +13,7 @@
 *** xref:how-to-guides/testing_applications/surface-level_tests.adoc[Surface-level tests]
 *** xref:how-to-guides/testing_applications/proc_adding_an_integration_test.adoc[Adding an integration test]
 *** xref:how-to-guides/testing_applications/proc_creating_custom_test.adoc[Creating a custom integration test]
+*** xref:pages/how-to-guides/testing_applications/proc_retriggering_integration_tests.adoc[Retriggering integration tests]
```
Comment (ralphbean) [inline docs/modules/ROOT/nav-how-to-guides.adoc:16]: See how the links above this all just link to `xref:how-to-guides/...`? I think you need to nix that `pages/` prefix:

```suggestion
*** xref:how-to-guides/testing_applications/proc_retriggering_integration_tests.adoc[Retriggering integration tests]
```

Comment (ralphbean) [reply docs/modules/ROOT/nav-how-to-guides.adoc:16]: Oh, actually - I just checked the demo link and your change actually works as written. I don't know why!

**Review by ralphbean — APPROVED** [review]

File: docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_retriggering_integration_tests.adoc [hunk 6]
```diff
@@ -0,0 +1,35 @@
+= Retriggering Integration Tests
+
+Integration test scenarios for a given snapshot can be re-triggered by adding a label to the snapshot.
+
+== Prerequisites
+- The snapshot for which tests will be retriggered has completed its initial tests
```
Comment (gtrivedi88) [inline docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_retriggering_integration_tests.adoc:6]:
```suggestion
- The snapshot must complete all its initial tests before you trigger subsequent tests.
```

File: docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_retriggering_integration_tests.adoc [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+- CLI access to the OpenShift cluster in question.  For information on getting access to the CLI, see xref:../../getting-started/getting_started_in_cli.adoc[Getting started in CLI]
```
Comment (gtrivedi88) [inline docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_retriggering_integration_tests.adoc:7]:
```suggestion
- Ensure you have CLI access to the specific OpenShift cluster. For information on obtaining CLI access, refer to  xref:../../getting-started/getting_started_in_cli.adoc[Getting started in CLI]
```

File: docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_retriggering_integration_tests.adoc [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+- An up-to-date kubectl binary.  The `oc` binary may be used as well.
+
+== Procedure
+. Determine which IntegrationTestScenario should be re-run.  For a given snapshot, only one scenario can
```
Comment (gtrivedi88) [inline docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_retriggering_integration_tests.adoc:11]:
```suggestion
. Identify the IntegrationTestScenario that needs to be rerun.  For a given snapshot, you can rerun only one IntegrationTestScenario
```

File: docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_retriggering_integration_tests.adoc [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+be re-run at a time.
```
Comment (gtrivedi88) [inline docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_retriggering_integration_tests.adoc:12]:
```suggestion
be re-run at a time.
```

File: docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_retriggering_integration_tests.adoc [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+. Add the label `test.appstudio.openshift.io/run` to the snapshot. Set the value to the name of the scenario to be re-run
```
Comment (gtrivedi88) [inline docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_retriggering_integration_tests.adoc:13]:
```suggestion
. Label the snapshot with `test.appstudio.openshift.io/run`, assigning the scenario name you wish to rerun as its value.
```

File: docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_retriggering_integration_tests.adoc [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+[source]
+$ kubectl label [snapshot name] test.appstudio.openshift.io/run=[scenario name]
+
+. The tests will be re-triggered automatically.  Once they are re-triggered the label will be removed.  At this point you
```
Comment (gtrivedi88) [inline docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_retriggering_integration_tests.adoc:17]:
```suggestion
. The tests are re-triggered automatically.  Once they are re-triggered, the system removes the label, allowing you to apply a new label for a different scenario if you wish to test multiple scenarios.
```

File: docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_retriggering_integration_tests.adoc [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+may add the label again with a different scenario if you wish to re-test multiple scenarios.
```
Comment (gtrivedi88) [inline docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_retriggering_integration_tests.adoc:18]:
```suggestion

```

File: docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_retriggering_integration_tests.adoc [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+[source]
+$ tkn pipelinerun list                                                                                                                                                          main 
+NAME                            STARTED         DURATION   STATUS
```
Comment (gtrivedi88) [inline docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_retriggering_integration_tests.adoc:21]:
```suggestion

[Example Output]
NAME                            STARTED         DURATION   STATUS
```

File: docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_retriggering_integration_tests.adoc [hunk 14]
```diff
@@ -0,0 +1,35 @@
...
+integration-pipelinerun-jfrdb   4 seconds ago   ---        Running
+
+. If you run `tkn pipelinerun describe [pipelinerun]` you can see that the `appstudio.openshift.io/snapshot` and `test.appstudio.openshift.io/scenario` labels reference the correct snapshot and scenario.
```
Comment (gtrivedi88) [inline docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_retriggering_integration_tests.adoc:24]:
```suggestion
. To verify the tests, run `tkn pipelinerun describe [pipelinerun]` to ensure that the labels `appstudio.openshift.io/snapshot` and `test.appstudio.openshift.io/scenario` correctly reference the correct snapshot and scenario.
```

File: docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_retriggering_integration_tests.adoc [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+[source]
+$ tkn pipelinerun describe integration-pipelinerun-jfrdb                                                                                                                        main 
+Name:              integration-pipelinerun-jfrdb
```
Comment (gtrivedi88) [inline docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_retriggering_integration_tests.adoc:27]:
```suggestion

[Expected labels in the output]
Name:              integration-pipelinerun-jfrdb
```

File: docs/modules/ROOT/nav-how-to-guides.adoc (hunk 5 above)
Comment (14rcole) [reply docs/modules/ROOT/nav-how-to-guides.adoc:16]: That's interesting, seems like `pages` is required for links to work in github but not in the site itself.  I'll revert that change.

File: docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_retriggering_integration_tests.adoc (hunk 6 above)
Comment (14rcole) [reply docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_retriggering_integration_tests.adoc:7]: I tweaked this suggestion a tiny bit, let me know if its fine this way.

File: docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_retriggering_integration_tests.adoc (hunk 14 above)
Comment (14rcole) [reply docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_retriggering_integration_tests.adoc]: I tweaked this suggestion a bit, let me know if its fine this way.

**Review by gtrivedi88 — APPROVED** [review]
LGTM

**Pull request merged**